src/utils/Processing.ts

```diff
@@ -59,8 +60,16 @@ export function ensureOutputDir(input: string, output: string, isIgPubContext: b
 export function readConfig(input: string): Configuration {
   const configPath = ensureConfiguration(input);
   if (configPath == null || !fs.existsSync(configPath)) {
-    logger.error('No config.yaml in FSH definition folder.');
-    throw Error;
+    const config = extractConfiguration(input);
```

Comment:
I thought this made sense, but looking a little deeper I'm less sure. It seems to me that `ensureConfiguration` ensures that a `config.yaml` file will exist. But when we extract the configuration, we don't actually make a `config.yaml` file, we are just directly creating the configuration from the ImplementationGuide JSON. I could still put this code in `ensureConfiguration`, but it feels a little off, since one method of ensuring the configuration, `generateConfiguration`, creates a file that is then read into memory, where the other method of ensuring the configuration `extractConfiguration`, generates a config directly in memory. Let me know what you think.

Reply:
Ugh.  You're right -- it returns a config _path_ -- not a config itself.  Ugh.  What do you think of this?
* Rename `ensureConfiguration` to `ensureConfigurationFile`
* Rename `extractConfiguration` to `loadConfigurationFromIgJSON`

I think that clarifies things a bit better...

Reply:
That sounds good to me!

Reply:
Ok, https://github.com/FHIR/sushi/pull/577/commits/2282a68eab01c5a16b6e3957582c6260bcb68ca3 changes the names, I went with `loadConfigurationFromIgResource`, since the IG can now be either JSON or XML.

Reply:
Perfect, especially since that avoids the whole "should it be 'JSON' or 'Json' debacle".

Reply:
'JSON' vs 'Json' and 'FHIR' vs 'Fhir' are two of life's unending debacles.